scripts/dependencies/lima.ts

```diff
@@ -0,0 +1,103 @@
+  githubOwner = 'rancher-sandbox';
+  githubRepo = 'lima-and-qemu';
+
+  async download(context: DownloadContext): Promise<void> {
+    const baseUrl = `https://github.com/${ this.githubOwner }/${ this.githubRepo }/releases/download`;
+    let platform: string = context.platform;
+
+    if (platform === 'darwin') {
+      platform = 'macos';
+      if (process.env.M1) {
+        platform = `macos-aarch64`;
+      }
+    }
+    const url = `${ baseUrl }/v${ context.versions.limaAndQemu }/lima-and-qemu.${ platform }.tar.gz`;
+    const expectedChecksum = (await getResource(`${ url }.sha512sum`)).split(/\s+/)[0];
+    const limaDir = path.join(context.resourcesDir, context.platform, 'lima');
+    const tarPath = path.join(context.resourcesDir, context.platform, `lima-v${ context.versions.limaAndQemu }.tgz`);
+
+    await download(url, tarPath, {
+      expectedChecksum, checksumAlgorithm: 'sha512', access: fs.constants.W_OK,
+    });
+    await fs.promises.mkdir(limaDir, { recursive: true });
+
+    const child = childProcess.spawn('/usr/bin/tar', ['-xf', tarPath],
+      { cwd: limaDir, stdio: 'inherit' });
+
+    await new Promise<void>((resolve, reject) => {
+      child.on('exit', (code, signal) => {
+        if (code === 0) {
+          resolve();
+        } else {
+          reject(new Error(`Lima extract failed with ${ code || signal }`));
+        }
+      });
+    });
+  }
+}
+
+export class AlpineLimaISO implements Dependency {
+  name = 'alpineLimaISO';
+  githubOwner = 'lima-vm';
+  githubRepo = 'alpine-lima';
+
+  async download(context: DownloadContext): Promise<void> {
+    const baseUrl = `https://github.com/${ this.githubOwner }/${ this.githubRepo }/releases/download`;
+    const edition = 'rd';
+    const version = context.versions.alpineLimaISO;
+    let arch = 'x86_64';
+
+    if (context.platform === 'darwin' && process.env.M1) {
+      arch = 'aarch64';
+    }
+    const isoName = `alpine-lima-${ edition }-${ version.alpineVersion }-${ arch }.iso`;
+    const url = `${ baseUrl }/v${ version.isoVersion }/${ isoName }`;
+    const destPath = path.join(process.cwd(), 'resources', os.platform(), `alpine-lima-v${ version.isoVersion }-${ edition }-${ version.alpineVersion }.iso`);
+    const expectedChecksum = (await getResource(`${ url }.sha512sum`)).split(/\s+/)[0];
+
+    await download(url, destPath, {
+      expectedChecksum, checksumAlgorithm: 'sha512', access: fs.constants.W_OK,
+    });
+  }
+
+  async getLatestVersion(): Promise<AlpineLimaISOVersion> {
+    // get latest isoVersion
+    const response = await getOctokit().rest.repos.listReleases({ owner: this.githubOwner, repo: this.githubRepo });
+    const latestRelease = response.data[0];
+    const latestVersionWithV = latestRelease.tag_name;
+    const isoVersion = latestVersionWithV.replace('v', '');
+
+    // get latest alpineVersion by parsing name of an asset on latest release
+    const matchingAsset = latestRelease.assets.find(asset => asset.name.includes('rd'));
```

Comment:
> error TS7006: Parameter 'asset' implicitly has an 'any' type.

Reply:
I think this happens when one doesn't run `npm install` after switching to the branch (I added the `octokit` dependency, which defines this).

![image](https://user-images.githubusercontent.com/19395776/188204271-af600119-459b-4147-b67e-7080cbbcc36d.png)